addons/stock/models/stock_move.py

```diff
@@ -557,6 +557,9 @@ def write(self, vals):
             for move in self.filtered(lambda m: m.state not in ('done', 'draft') and m.picking_id):
                 if float_compare(vals['product_uom_qty'], move.product_uom_qty, precision_rounding=move.product_uom.rounding):
                     self.env['stock.move.line']._log_message(move.picking_id, move, 'stock.track_move_template', vals)
+                    if move.picking_id.immediate_transfer and move.move_dest_ids:
```

Comment:
Okay, the use case seems legit since we also increase the initial demand on extra move. However I think the fix is not at the good place. Could you check why _create_extra_move does not trigger the rule with the quantity done on the new move. Also why does the push rules are trigger with qty = 0

Reply:
`_create_extra_move` is called in `_action_done` whereas `_push_apply` (which calls `_run_push`)  is called during `_action_confirm`. If I understand it correctly then there's currently no way for immediate transfers to trigger its next (picking) steps with the correct Demand  since they are confirmed as soon as they are saved (and we currently can't add Done values via the action_show_details widget until its saved). Do we want to make it so `_push_apply` is called for immediate transfers during `_action_done` instead of `_action_confirm`?

Reply:
@ticodoo I would say that the first _action_confirm should not call _run_push since it doesn't have initial demand. But instead when it's validated, it should create the extra move with the extra quantity as initial demand and then call call _action_confirm and _run_push on it.

Reply:
I added a fixup where immediate transfers does its `_run_push` after it's validated rather than when it's first saved. I left it as is for planned transfers so the current behavior doesn't change. What do you think?